kotlinx-coroutines-core/common/src/Annotations.kt

```diff
@@ -6,6 +6,18 @@ package kotlinx.coroutines
 
 import kotlinx.coroutines.flow.*
 
+/**
+ * Marks declarations in the coroutines that are **delicate** &mdash;
+ * they have limited use-case and shall be used with care in general code.
+ * Any use of a delicate declaration has to be carefully reviewed to make sure it is
+ * properly used and does not create problems like memory and resource leaks.
+ * Carefully read documentation of any declaration marked as `DelicateCoroutinesApi`.
+ */
+@MustBeDocumented
+@Retention(value = AnnotationRetention.BINARY)
+@RequiresOptIn(level = RequiresOptIn.Level.WARNING)
+public annotation class DelicateCoroutinesApi
```

Comment:
Please specify the custom warning message for `DelicateCoroutinesApi` that suggests that is is safe to opt-in as long as one read all the corresponding documentation